src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -161,15 +160,8 @@ internal static int Start(string bannerText, string helpText, string[] args)
                     hostException = e;
                 }
 
-                PSHostUserInterface hostUi = s_theConsoleHost?.UI ?? new NullHostUserInterface();
-                s_cpp = new CommandLineParameterParser(hostUi, bannerText, helpText);
-                s_cpp.Parse(args);
-
-#if UNIX
-                // On Unix, logging has to be deferred until after command-line parsing
-                // completes to allow overriding logging options.
-                PSEtwLog.LogConsoleStartup();
-#endif
+                PSHostUserInterface hostUI = s_theConsoleHost?.UI ?? new NullHostUserInterface();
+                s_cpp.ShowErrorHelpBanner(hostUI, bannerText, helpText, args);
```

Comment:
Just found a big problem. Any custom host that use `ConsoleShell.Start` as the entry point won't work anymore because the `args` won't be parsed in `ConsoleShell.Start`. This requirement basically means the argument parsing has to happen in `ConsoleShell.Start`.

Reply:
Nice catch! I skipped that it is a public API. 

Have you a concern about PSEtwLog.LogConsoleStartup();?